src/core/server/logging/ecs.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * and the Server Side Public License, v 1; you may not use this file except in
+ * compliance with, at your election, the Elastic License or the Server Side
+ * Public License, v 1.
+ */
+
+/**
+ * Typings for some ECS fields which core uses internally.
+ * These are not a complete set of ECS typings and should not
+ * be used externally; the only types included here are ones
+ * currently used in core.
+ *
+ * @internal
+ */
+
+export interface EcsOpsMetricsEvent {
+  /**
+   * These typings were written as of ECS 1.7.0.
+   * Don't change this value without checking the rest
+   * of the types to conform to that ECS version.
+   *
+   * https://www.elastic.co/guide/en/ecs/1.7/index.html
+   */
+  ecs: { version: '1.7.0' };
+
+  // base fields
+  ['@timestamp']?: string;
+  labels?: Record<string, unknown>;
+  message?: string;
+  tags?: string[];
+  // other fields
+  process?: EcsProcessField;
+  event?: EcsEventField;
+}
+
+interface EcsProcessField {
+  uptime?: number;
+}
+
+interface EcsEventField {
+  kind?: string;
+  category?: string[];
+  type?: string;
```

Comment:
nit: [kind](https://www.elastic.co/guide/en/ecs/current/ecs-allowed-values-event-kind.html#ecs-allowed-values-event-kind), [category](https://www.elastic.co/guide/en/ecs/current/ecs-allowed-values-event-category.html#ecs-allowed-values-event-category), and [type](https://www.elastic.co/guide/en/ecs/current/ecs-allowed-values-event-type.html#ecs-allowed-values-event-type) all have a fixed list of allowed values. We should consider explicitly typing those here instead of using `string`.

Reply:
@lukeelmers That's a great idea and actually not a nit since we're trying to be as compatible with ECS 1.7.0 as possible.
I gave [it a shot](https://github.com/elastic/kibana/pull/88070/files#diff-5a63ef9defe0307279013891ae6412d539b0d2c72500634815964f15420e3e38R43-R45) and we can iterate/improve on the typings as we go.